lib/semantics/assignment.cpp

```diff
@@ -333,14 +218,68 @@ void AssignmentContext::CheckForPureContext(const SomeExpr &lhs,
   }
 }
 
-MaskExpr AssignmentContext::GetMask(
-    const parser::LogicalExpr &logicalExpr, bool defaultValue) {
-  MaskExpr mask{defaultValue};
-  if (const SomeExpr * expr{GetExpr(logicalExpr)}) {
-    auto *logical{std::get_if<evaluate::Expr<evaluate::SomeLogical>>(&expr->u)};
-    mask = evaluate::ConvertTo(mask, common::Clone(DEREF(logical)));
+// 10.2.3.1(2) The masks and LHS of assignments must all have the same shape
+void AssignmentContext::CheckShape(parser::CharBlock at, const SomeExpr *expr) {
+  if (auto shape{evaluate::GetShape(foldingContext(), expr)}) {
+    std::size_t size{shape->size()};
+    if (whereDepth_ == 0) {
+      whereExtents_.resize(size);
+    } else if (whereExtents_.size() != size) {
+      Say(at,
+          "Must have rank %jd to match prior mask or assignment of"
```

Comment:
`%zd` is for `size_t`; `%jd` is for `intmax_t`.

Reply:
OK